### ISSUE TITLE
Powerwall not showing

### ISSUE THREAD

**Issue by olej24** [alandtse/tesla#451]
Hi

I've installed the latest version and dev version and my Model Y works fine, but the integration isn't finding my Powerwall.

Any suggestions?

**Comment by InTheDaylight14** [comment]
Double check the "Include Energy Sites" is checked off when setting up the integration.

Also is the power wall under the same tesla account?

![image](https://user-images.githubusercontent.com/67400055/208959051-a6c8acc9-2900-4986-a955-18759a459995.png)

**Comment by olej24** [comment]
Yes, both are showing on the same account in the Tesla app. The PW was only commissioned this week so the only thing i can think is that it takes some time to show up?

**Comment by craigrouse** [comment]
@olej24 when mine was first commissioned, it took several weeks before I was able to control it with the app, even though I could see it in the app. It's quite possible this is the issue. Do you have access in the Tesla app to change the operation settings (e.g. Time-based Control, Self-Powered)?

**Comment by olej24** [comment]
Hi @craigrouse ? Did you mean “it took several weeks befor I was able to control it with home assistant”? 
yes to be clear, once commissioned it showed up in my Tesla iPhone app straight away, along with existing car. But it hasn’t showed up on the HA integration, the car does show up in HA.

**Comment by InTheDaylight14** [comment]
> Hi @craigrouse ? Did you mean “it took several weeks befor I was able to control it with home assistant”? yes to be clear, once commissioned it showed up in my Tesla iPhone app straight away and I’m able to control it along with existing car. But it hasn’t showed up on the HA integration, the car does show up in HA.

Wait, have you removed and then re-added the Tesla custom integration in HA? You will need to do that for the Powerwall to be setup

**Comment by olej24** [comment]
Yes, tried that. No help

**Comment by Twiederh** [comment]
Hi,
I have the same issue, I just tried to configure a fresh home assistant with the custom Tesla integration. Energy sites is checked but after the initialization of the custom Tesla plugin, only my Model 3 shows up. I also contacted Tesla energy support, they say they do not support the API and that the powerwall shows up under the account I am using for the custom integration (which is my main account anyway).
Has anyone yet solved this issue?
